tests/slurm_command_gen_strategy/test_ucc_slurm_command_gen_strategy.py

```diff
@@ -59,13 +59,39 @@ def cmd_gen_strategy(self, slurm_system: SlurmSystem) -> UCCTestSlurmCommandGenS
     )
     def test_generate_test_command(
         self,
+        tmp_path: Path,
         cmd_gen_strategy: UCCTestSlurmCommandGenStrategy,
-        cmd_args: Dict[str, Union[str, List[str]]],
-        extra_cmd_args: str,
-        expected_command: List[str],
+        cmd_args_data: dict,
+        expected_command: list[str],
     ) -> None:
-        env_vars = {}
-        tr = Mock()
-        tr.test.extra_cmd_args = extra_cmd_args
-        command = cmd_gen_strategy.generate_test_command(env_vars, cmd_args, tr)
+        ucc_cmd_args = UCCCmdArgs(
+            collective=cmd_args_data["collective"],
+            b=cmd_args_data["b"],
+            e=cmd_args_data.get("e", "8M"),
+        )
+
+        test_def = UCCTestDefinition(
+            name="ucc_test",
+            description="UCC test",
+            test_template_name="default_template",
+            cmd_args=ucc_cmd_args,
+            extra_env_vars={},
+            extra_cmd_args={},
+        )
+
+        test_obj = Test(test_definition=test_def, test_template=Mock())
+
+        tr = TestRun(
+            test=test_obj,
+            num_nodes=1,
+            nodes=[],
+            output_path=tmp_path / "output",
+            name="test-job",
+        )
+
+        command = cmd_gen_strategy.generate_test_command(
+            test_def.extra_env_vars,
+            test_def.cmd_args.model_dump(),
```

Comment:
IMO it should be `test_def.cmd_args_dict`, we might have some logic inside this method.